static/src/javascripts/projects/common/utils/steady-page.js

```diff
@@ -0,0 +1,187 @@
+/*
+ Utility: steady-insert.js
+ Insert components into web-pages without jumping them around by
+ moving the current scroll position by the same distance as the height of the
+ inserted element.
+ Note: Inserted elements must have a fixed height when rendered
+ (i.e: no changing the height by loading an image, lazy-loading content etc)
+ */
+
+define([
+    'common/utils/fastdom-promise',
+    'Promise',
+    'lodash/objects/assign',
+    'common/utils/queue',
+    'common/utils/config'
+], function (
+    fastdom,
+    Promise,
+    assign,
+    Queue,
+    config
+) {
+
+    var q = new Queue();
+    var running = false;
+    var promise;
+
+    /**
+     * Process the insertion operation:
+     *
+     *   1. Calculate the original height of the container
+     *   2. Apply the insertion functions for all inserted elements
+     *   3. Calculate the new height of the container
+     *   4. Adjust the scroll position to account for the new container height
+     *
+     * @param  {Number} state.scrollY The original scroll position
+     */
+    function go(state) {
+        running = true;
+
+        var batch = [];
+        var batchHeightsBeforeInsert;
+
+        promise = fastdom.read(function(){
```

Comment:
As far as I can see, there is no DOM-related ops in this routine, so you could create a `Promise` instead of calling `fastdom.read`, no?